server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -127,18 +140,30 @@ public SourceFieldMapper build() {
                 throw new MapperParsingException("Cannot set both [mode] and [enabled] parameters");
             }
             if (isDefault()) {
-                return DEFAULT;
+                return indexMode == IndexMode.TIME_SERIES ? TSDB_DEFAULT : DEFAULT;
+            }
+            if (indexMode == IndexMode.TIME_SERIES) {
```

Comment:
I'd prefer to have a method on `IndexMode` that does this stuff if possible.

Reply:
Turns out that the usage of include and exclude in combination with synthetic source is already not allowed by validation in `SourceFieldMapper`'s constructor.

I'll just validate that here that synthetic source is enabled.